docs/notebooks/multiobjective_ehvi.pct.py

```diff
@@ -0,0 +1,170 @@
+plt.show()
+
+
+# ## Modelling the two functions
+#
+# We'll model the different objective functions individually their own Gaussian process regression models.
+
+
+def create_bo_model(data, input_dim=2, l=1.0):
+    variance = tf.math.reduce_variance(data.observations)
+    lengthscale = l * np.ones(input_dim, dtype=gpflow.default_float())
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=lengthscale)
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+    return create_model(
+        {
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    )
+
+
+objective_models = [
+    (
+        create_bo_model(
+            Dataset(
+                initial_data[OBJECTIVE].query_points,
+                tf.gather(initial_data[OBJECTIVE].observations, [i], axis=1),
+            )
+        ),
+        1,
+    )
+    for i in range(num_objective)
+]
+
+# And stack the two independent GP model in a single `ModelStack` multioutput model representing predictions through different outputs. 
+
+models = {OBJECTIVE: ModelStack(*objective_models)}
+
+# ## Define the acquisition process
+#
+# Here we utilize the [EHVI](https://link.springer.com/article/10.1007/s10898-019-00798-7): `ExpectedHypervolumeImprovement` acquisition function:
+
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+ehvi = ExpectedHypervolumeImprovement()
+rule: EfficientGlobalOptimization[Box] = EfficientGlobalOptimization(builder=ehvi.using(OBJECTIVE))
+
+# ## Run the optimization loop
+#
+# We can now run the optimization loop
+
+num_steps = 20
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+result = bo.optimize(num_steps, initial_data, models, acquisition_rule=rule)
+
+# To conclude, we visualize the multi-objective bo queried data in the design space
+
+# +
+datasets = result.try_get_final_datasets()
+data_query_points = datasets[OBJECTIVE].query_points.numpy()
+data_observations = datasets[OBJECTIVE].observations.numpy()
+
+_, ax = plot_function_2d(
+    vlmop2, mins, maxs, grid_density=100, contour=True, title=["Obj 1", "Obj 2"]
+)
+plot_bo_points(data_query_points, ax=ax[0, 0], num_init=num_initial_points)
+plot_bo_points(data_query_points, ax=ax[0, 1], num_init=num_initial_points)
+plt.show()
+# -
+
+# ... and visulize in the objective space, purple dots denotes the nondominated points.
+
+plot_bo_points_in_obj_space(data_observations, num_init=num_initial_points)
+plt.show()
+
+# We can also visualize how the performance metric get envolved with respect to the bo iterations.  
+# First, we need to define a metric function by ourself. The performance metric function for multi-objective function can be fleaxibly choosen. Here, we make use of the log hypervolume difference.
```

Comment:
First, we need to define a performance metric. Many metrics have been considered for multi-objective function. Here, we use the log hypervolume difference.